### DISCUSSION TITLE
Scaling an axis of a mesh

### DISCUSSION THREAD

**Discussion by ghost** [pyvista/pyvista#6642] [Q&A] [answered]
### First check

- [ ] I added a very descriptive title here.
- [x] I used the GitHub search to find a similar question and didn't find it.
- [x] I searched the PyVista documentation, with the integrated search.
- [x] I already searched in Google "How to X in PyVista" and didn't find any information.
- [ ] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I will NOT make duplicate discussions or issues in the future.

### Commit to Help

- [X] I commit to help with one of those options.

### Sample Code What is the problem, question, or error?
Write a short description telling me what you are doing, what you expect to happen, and what is currently happening.

I have the same problem like https://github.com/pyvista/pyvista/discussions/4260, I want to scale the mesh properly and show the axis value correctly which means axis should show the true value instead of scaled value. Can I do it in another method?

### Description

```python
import numpy as np
import pyvista as pv

# Make data
x = np.arange(-10, 10, 0.25)
y = np.arange(-1000, 1000, 2.5)
x, y = np.meshgrid(x, y)
r = np.sqrt(x ** 2 + (y / 100) ** 2)
z = 10000 * np.sin(r)

pl = pv.Plotter()
pl.set_scale(yscale=0.01, zscale=0.0001)
pl.add_mesh(pv.StructuredGrid(x, y, z), scalars=z.T.flatten())
pl.show_grid()
pl.show()
```
![image](https://github.com/user-attachments/assets/1dd78401-16c9-4bdf-a490-0058b47bde8c)
for example, in case above, can i show the axis of y range from -1000 to 1000 and z from -10000 to 10000?

### System Information

```shell
OS : Windows
            CPU(s) : 12
           Machine : AMD64
      Architecture : 64bit
               RAM : 15.8 GiB
       Environment : Python
        GPU Vendor : Intel
      GPU Renderer : Intel(R) UHD Graphics 630
       GPU Version : 4.5.0 - Build 31.0.101.2111
  MathText Support : False

  Python 3.10.8 (tags/v3.10.8:aaaf517, Oct 11 2022, 16:50:30) [MSC v.1933 64
  bit (AMD64)]

           pyvista : 0.43.4
               vtk : 9.3.1
             numpy : 1.25.0
        matplotlib : 3.7.1
            scooby : 0.7.2
             pooch : v1.7.0
            pillow : 9.5.0
           imageio : 2.34.0
         pyvistaqt : 0.11.0
             PyQt5 : 5.15.10
           IPython : 8.20.0
          colorcet : 3.0.1
        ipywidgets : 8.1.1
             scipy : 1.14.0
              tqdm : 4.66.1
            meshio : 5.3.5
```

**Comment by darikg** [comment, marked as answer]
can't figure out how to set a scale on the grid but you could set them manually

```python
grid = pl.show_grid()
new_labels = pv.core.utilities.arrays.convert_string_array(np.array(['-1000', '-500', '0', '500', '1000']))
grid.SetAxisLabels(1, new_labels)  # y-axis is axis 1, z is 2, etc
```

**Reply by ghost** [reply]
Thanks for your reply, the code works. I use grid.x_labels/y_labels/z_labels to get the axis value and calculate the true value.

**Reply by tkoyama010** [reply]
@darikg Thank you. Would you give your thought about this should be added to the PyVista methods?

**Reply by darikg** [reply]
looking at `pv.CubeAxesActor` I see two potential feature additions:

1. Add `@x_labels.setter` etc property setters to handle converting to a vtkStringArray and dispatching to `SetAxisLabels` for any custom tick label overrides

2. Add `x_scale` etc properties to pass to `pyvista.plotting.cube_axes_actor.make_axis_labels` to avoid needing to calculate the custom scaled values
